### ISSUE TITLE
Keep Hitting Data not formatted properly 

### ISSUE THREAD

**Issue by simonzcaiman** [rhiever/tpot#468]
Here is my code : 
    
    
x=np.array([[-1.1491834 , -1.77252632,  0.72741207],
       [-0.69059757, -1.59468535,  0.13948231],
       [ 0.06890373,  0.43569532,  0.21672055],
       [ 1.77999225,  0.23434428,  0.49602475],
       [ 1.09255898,  1.56640373, -0.47355987]])

y=array([ 1.,  1.,  0., -1.,  0.])
 
 
 
 tpot = TPOTClassifier(generations=2, population_size=30, verbosity=2,cv=5,scoring='accuracy')
 tpot.fit(X_train[:5], y_train[:5])
 

  tpot = TPOTClassifier(generations=2, population_size=30, verbosity=2,cv=5,scoring='average_precision')
 tpot.fit(X_train[:5], y_train[:5])

Console message: 
    
'''
Optimization Progress:  67%|██████▋   | 60/90 [00:04<00:04,  7.13pipeline/s]Generation 1 - Current best internal CV score: inf
                                                                            Generation 2 - Current best internal CV score: inf
There was an error in the TPOT optimization process. This could be because the data was not formatted properly, or because data for a regression problem was provided to the TPOTClassifier object. Please make sure you passed the data to TPOT correctly.
'''

Is there anything wrong with my dateset?

**Comment by weixuanfu2016** [comment]
Hmm, I think the reason is the sample size is too small to use `cv=5` in TPOT. Please check the cv parameter in [`cross_eval_score` in scikit-learn](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html)

The codes below can reproduce this issue.

```
import numpy as np
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score
clf = DecisionTreeClassifier(max_depth=5)
y=np.array([ 1., 1., 0., -1., 0.])
x=np.array([[-1.1491834 , -1.77252632, 0.72741207],
[-0.69059757, -1.59468535, 0.13948231],
[ 0.06890373, 0.43569532, 0.21672055],
[ 1.77999225, 0.23434428, 0.49602475],
[ 1.09255898, 1.56640373, -0.47355987]])
# pass with warning
cv_scores_1 = cross_val_score(clf, x, y, cv=2)
print(cv_scores_1)
# fail here
cv_scores_2 = cross_val_score(clf, x, y, cv=5)
```

**Comment by rhiever** [comment]
@weixuanfu2016 is correct. If you are using `cv=5`, you must have at least 5 training samples for each class.

**Comment by simonzcaiman** [comment]
Thanks guys. using cv=2 will solve the puzzle. 
if one decides to use " roc_auc" score as the scoring function for TPOTClassifier object, 
the internal CV will return "inf", when there is an exception occurred in the scoring function. 
I wrote a sample code that can reproduce the error. It is because the roc_auc can only handle binary class labels. It will raise error for multiclasses. 
However, tpot currently is not passing any error message. Maybe passing the error message from sklearn function?
 

import sklearn.metrics.roc_auc_score as roc
metric.roc_auc_score([1,-1,1],[1,-1,1])
